### PR TITLE
Add an improved nightly release pipeline

### PR DESCRIPTION
**Pull request by nikhil-thomas**
Add a nighlty release pipeline which can fetch nightly releases of
components before making operator nightly release

Add a Task which can fetch nightly releases of Pipeline, Trigger and Dashboard

Signed-off-by: Nikhil Thomas <nikthoma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by nikhil-thomas** [comment]
/hold
"kustomizing for plumbing"

File: tekton/overlays/nightly-releases/fetch-nightly-components.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      #!/usr/bin/env sh
+      set -ex
+      cd /go/src/github.com/tektoncd/operator
+      components="pipeline trigger dashboard"
+      for component in ${components}; do
+        componentPath=cmd/kubernetes/operator/kodata/tekton-${component}
+        rm -rf ${componentPath}/*
+        mkdir -p ${componentPath}/nightly
+        if [[ ${component} == "dashboard" ]]; then
+          curl -L -o ${componentPath}/nightly/release.notags.yaml https://storage.googleapis.com/tekton-releases-nightly/${component}/latest/tekton-dashboard-release.yaml
```
Comment (afrittoli) [inline tekton/overlays/nightly-releases/fetch-nightly-components.yaml:24]: Note that in some case the nightly build from a component may have failed, so the latest across components may belong to different days. I think that's perfectly acceptable, I just wanted to highlight it.

File: tekton/overlays/nightly-releases/fetch-nightly-components.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          wc -l ${componentPath}/nightly/release.notags.yaml
```
Comment (afrittoli) [inline tekton/overlays/nightly-releases/fetch-nightly-components.yaml:25]: This looks as if it was added for debugging purposed.
If it's something that you want to leave in the final pipeline perhaps you could add a nice message to it?
```
INITIAL_COUNT=$(wc -l ${componentPath}/nightly/release.notags.yaml)
sed -i '/aggregationRule/,+3d' ${componentPath}/nightly/release.notags.yaml
          wc -l ${componentPath}/nightly/release.notags.yaml
FINAL_COUNT=$(wc -l ${componentPath}/nightly/release.notags.yaml)
echo "Dashboard release line modified from $INITIAL_COUNT to $FINAL_COUNT"
```

File: tekton/overlays/nightly-releases/fetch-nightly-components.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          sed -i '/aggregationRule/,+3d' ${componentPath}/nightly/release.notags.yaml
+          wc -l ${componentPath}/nightly/release.notags.yaml
+          continue
+        fi
+        curl -L -o ${componentPath}/nightly/release.notags.yaml https://storage.googleapis.com/tekton-releases-nightly/${component}/latest/release.notags.yaml
```
Comment (afrittoli) [inline tekton/overlays/nightly-releases/fetch-nightly-components.yaml:30]: I wonder if we could have a param/flag in this task about using `notags` or not? 
Any reason for fetching only the notags version of the releases?

**Review by afrittoli — COMMENTED** [review]
Nice!
I've not reviewed the whole PR yet, just a couple of minor comments reading the new task :)

File: tekton/overlays/nightly-releases/fetch-nightly-components.yaml (hunk 1 above)
Comment (nikhil-thomas) [reply tekton/overlays/nightly-releases/fetch-nightly-components.yaml:24]: true. 👍 

i shall make sure that we communitcate this clearly in the operator README.

i shall add somethingling like
```
in some case the nightly build from a component may have failed, so the latest across components may belong to different days.
```

File: tekton/overlays/nightly-releases/fetch-nightly-components.yaml (hunk 3 above)
Comment (nikhil-thomas) [reply tekton/overlays/nightly-releases/fetch-nightly-components.yaml:28]: ooops 😛 
fixed 👍

File: tekton/overlays/nightly-releases/fetch-nightly-components.yaml (hunk 2 above)
Comment (nikhil-thomas) [reply tekton/overlays/nightly-releases/fetch-nightly-components.yaml]: ooops 😛 
fixed 👍

File: tekton/overlays/nightly-releases/fetch-nightly-components.yaml (hunk 3 above)
Comment (nikhil-thomas) [reply tekton/overlays/nightly-releases/fetch-nightly-components.yaml:28]: few months back notags seem to work on all kubernetes platforms.
i shall discuss this in today's operator workgroup meeting and make call.

if we decide to add a param, then i shall make another pr. 🧑‍💻

Comment (vdemeester) [reply tekton/overlays/nightly-releases/fetch-nightly-components.yaml:28]: @afrittoli the only reason for using `notags` is to be sure it works with all kubernetes CRI implementation 😝.

**Review by vdemeester — APPROVED** [review]
Nice @nikhil-thomas 
/meow

**Pull request merged**